gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -840,38 +840,40 @@ function TellTraitorsAboutTraitors()
 			traitornicks[#traitornicks + 1] = v:Nick()
 		end
 	end
+	
+	for _, v in ipairs(player.GetAll()) do
+		if v:HasTeam(TEAM_TRAITOR) then
 
-	hook.Run("TTT2TellTraitors", traitornicks)
+			local tmp, shouldShow = hook.Run("TTT2TellTraitors", v, table.Copy(traitornicks))
```

Comment:
I suggest to not breaking any other add-ons. To do so, you should change this into
```lua
local shouldShow = hook.Run("TTT2TellTraitors", traitornicks, v)
```

traitornicks should be a copy for each player that is created before calling this hook. Maybe the structure needs to change, but this will support the integrity of already existing add-ons.

Reply:
So should I just move the creation of the traitornicks table into the for loop of the players?

Reply:
Just creating a new table as a copy of traitornicks